src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -214,7 +220,12 @@
   [ca-ttl :- schema/Int]
   (let [now        (time/now)
         not-before (time/minus now (time/days 1))
-        not-after  (time/plus now (time/seconds ca-ttl))]
+        ;; clj-time specific interval classes (eg seconds) do not
+        ;; accept inputs larger than int, so we cast to a long
+        ;; representation of millis and construct our own Period instance
+        ;; for time/plus
```

Comment:
It seems very non-obvious to me why we're doing this. And so I felt that it needed a comment, but if felt a bit dirty adding the comment to otherwise clean code. Should I put this information in a different form or is it more obvious to folks who are more familiar with the clj ecosystem?